apps/src/codegen.js

```diff
@@ -52,20 +52,20 @@ exports.evalWith = function (code, options, legacy) {
  * Generate code for each of the given events, and evaluate it using the
  * provided APIs as context.
  * @param apis Context to be set as globals in the interpreted runtime.
- * @param events Mapping of hook names to the corresponding blockly block name.
- * @param generator Function for generating code for a given block name.
+ * @param events Mapping of hook names to the corresponding handler code.
+ * @param code Optional extra code to evaluate.
  * @return {{}} Mapping of hook names to the corresponding event handler.
  */
-exports.evalWithEvents = function (apis, events, generator) {
+exports.evalWithEvents = function (apis, events, code) {
   let interpreter, currentCallback;
-  let code = '';
+  code = code || '';
```

Comment:
I've updated this to use an optional param in https://github.com/code-dot-org/code-dot-org/pull/10363/files#diff-11d38f100f6df9c907f72e0cf6bb3671R59.